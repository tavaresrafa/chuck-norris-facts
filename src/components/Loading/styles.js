import styled, { keyframes } from 'styled-components';
import { colors } from '../../styles/colors';

const bounceAnimation = keyframes`
  0%, 80%, 100% {transform: scale(0);}
  40% {transform: scale(1.0);}
`;

export const Loading = styled.div`
  margin: 100px auto 0;
  width: 70px;
  text-align: center;

  > div {
    width: 18px;
    height: 18px;
    background-color: ${colors.orange};
    border-radius: 100%;
    display: inline-block;
    animation: ${bounceAnimation} 1.4s infinite ease-in-out both;
  }

  .bounce1 {
    animation-delay: -0.32s;
  }

  .bounce2 {
    animation-delay: -0.16s;
  }
`;
