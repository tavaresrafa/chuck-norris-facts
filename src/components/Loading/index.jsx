import React from 'react';
import * as S from './styles';

const Loading = () => (
  <S.Loading>
    <div data-testid="bounce1" className="bounce1"></div>
    <div data-testid="bounce2" className="bounce2"></div>
    <div data-testid="bounce3" className="bounce3"></div>
  </S.Loading>
)

export default Loading;
