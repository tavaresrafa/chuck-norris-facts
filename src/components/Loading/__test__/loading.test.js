import React from 'react';
import ReactDOM from 'react-dom';
import Loading from '../../Loading';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement("div");
    ReactDOM.render(<Loading />, div);
});

it('it has class', () => {
    const { getByTestId } = render(<Loading />);
    expect(getByTestId('bounce1')).toHaveClass('bounce1');
    expect(getByTestId('bounce2')).toHaveClass('bounce2');
    expect(getByTestId('bounce3')).toHaveClass('bounce3');
});

// SNAPSHOTS
it('matches snapshot', () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
});