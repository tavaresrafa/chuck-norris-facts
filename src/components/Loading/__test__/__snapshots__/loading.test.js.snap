// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-AxjAm grrbNP"
>
  <div
    className="bounce1"
    data-testid="bounce1"
  />
  <div
    className="bounce2"
    data-testid="bounce2"
  />
  <div
    className="bounce3"
    data-testid="bounce3"
  />
</div>
`;
